salt/modules/kubernetes.py

```diff
@@ -692,7 +715,26 @@ def delete_deployment(name, namespace='default', **kwargs):
             name=name,
             namespace=namespace,
             body=body)
-        return api_response.to_dict()
+        mutable_api_response = api_response.to_dict()
+        if not platform.startswith("win"):
+            try:
+                with _time_limit(_polling_time_limit):
+                    while show_deployment(name, namespace) is not None:  # pylint: disable=useless-else-on-loop
+                        sleep(1)
+                    else:
+                        mutable_api_response['code'] = 200
+            except TimeoutException:
+                pass
+        else:
+            # Windows has not signal.alarm implementation, so we are just falling
+            # back to loop-counting.
+            for i in range(60):
+                if show_deployment(name, namespace) is None:
+                    mutable_api_response['code'] = 200
+                    break
+                else:
+                    sleep(1)
+        return mutable_api_response
```

Comment:
Does it make sense to have the `log.warning` somewhere over here? Only when `mutable_api_response['code'] != 200` ?